### ISSUE TITLE
Revise integration tests to remove reliance on clock time

### ISSUE THREAD

**Issue by cpu** [letsencrypt/ct-woodpecker#25]
Right now the ct-woodpecker integration tests use a function called `woodpeckerRun` to start an instance of a `woodpecker.Woodpecker` that is allowed to run for a pre-defined amount of time based on the configured submission/sth fetch intervals before being stopped gracefully:
https://github.com/letsencrypt/ct-woodpecker/blob/046129d977ce3ec8ffc8f6ba4ed904a6dbe65a0e/test/integration_test.go#L73-L82

On my local system this is quite reliable & fast. We can set thresholds in the sub-second range and the tests seem solid. In Travis CI things are less nice: It seems like frequently the test driver kills the woodpecker instance after the configured time but the instance hasn't completed its work as expected. Similarly there are times when the test driver kills the woodpecker instance later than expected.

To work around the first case of flakyness (where the woodpecker is killed before all work is done) we have been adding "padding" to the run duration: https://github.com/letsencrypt/ct-woodpecker/blob/046129d977ce3ec8ffc8f6ba4ed904a6dbe65a0e/test/integration_test.go#L285-L287

That _mostly_ prevents flaky failures, but they do crop up on occasion. 

To work around the second case of flakyness (where the woodpecker is killed later than expected and more work has occurred than planned) we consider our expected values as minimums and allow overages without error. That makes our tests less robust than they could be otherwise.

Overall it feels like we should revisit this testing strategy and come up with something not influenced by clock time.

Perhaps a better alternative is for the `woodpecker.Woodpecker` to count how many STH fetch/cert submission "cycles" it has done and kill the test based on the reported cycles being == desired. Open to suggestions!

**Comment by cpu** [comment]
I was able to get the `cpu-submit-precerts` branch from #24 to fail on my laptop after 131 full test runs using `COUNT=0; until ! GOCACHE=off go test -tags=integration -race ./...; do COUNT=$((count+1)); echo "Trying again - $COUNT"; done`:
```
ct-test-srv ":4500" 2018/06/07 16:40:19 Running cttestsrv instance on :4500 with pubkey MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEYggOxPnPkzKBIhTacSYoIfnSL2jPugcbUKx83vFMvk5gKAz/AGe87w20riuPwEGn229hKVbEKHFB61NIqNHC3Q==
ct-test-srv ":4501" 2018/06/07 16:40:19 Running cttestsrv instance on :4501 with pubkey MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEKtnFevaXV/kB8dmhCNZHmxKVLcHX1plaAsY9LrKilhYxdmQZiu36LvAvosTsqMVqRK9a96nC8VaxAdaHUbM8EA==
ct-test-srv ":4500" 2018/06/07 16:40:19 Stopping server on :4500
ct-test-srv ":4501" 2018/06/07 16:40:19 Stopping server on :4501
ct-test-srv ":4500" 2018/06/07 16:40:19 :4500 ListenAndServe error: http: Server closed
ct-test-srv ":4501" 2018/06/07 16:40:19 :4501 ListenAndServe error: http: Server closed
ct-test-srv ":4500" 2018/06/07 16:40:19 Running cttestsrv instance on :4500 with pubkey MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEYggOxPnPkzKBIhTacSYoIfnSL2jPugcbUKx83vFMvk5gKAz/AGe87w20riuPwEGn229hKVbEKHFB61NIqNHC3Q==
ct-test-srv ":4501" 2018/06/07 16:40:19 Running cttestsrv instance on :4501 with pubkey MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEKtnFevaXV/kB8dmhCNZHmxKVLcHX1plaAsY9LrKilhYxdmQZiu36LvAvosTsqMVqRK9a96nC8VaxAdaHUbM8EA==
ct-test-srv ":4500" 2018/06/07 16:40:20 Stopping server on :4500
ct-test-srv ":4500" 2018/06/07 16:40:20 :4500 ListenAndServe error: http: Server closed
ct-test-srv ":4501" 2018/06/07 16:40:21 Stopping server on :4501
--- FAIL: TestCertSubmissionSuccess (1.41s)
	integration_test.go:335: expected cert_submit_results count of >= 3 for log :4500, found 2
	integration_test.go:350: expected cert_submit_latency_count of >= 3 for log :4500, found 2
ct-test-srv ":4501" 2018/06/07 16:40:21 :4501 ListenAndServe error: http: Server closed
FAIL
FAIL	github.com/letsencrypt/ct-woodpecker/test	1.737s
```

**Comment by cpu** [comment]
After discussion we went with the simpler solution in https://github.com/letsencrypt/ct-woodpecker/pull/30

**Comment by cpu** [comment]
Calling this fixed with #30 until evidence proves contrary.
